build.gradle.kts

```diff
@@ -28,6 +28,25 @@ subprojects {
             include("**/kotlin/**")
         }
     }
+
+    tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile>().configureEach {
+        sourceCompatibility = JavaVersion.VERSION_1_8.toString()
+        targetCompatibility = JavaVersion.VERSION_1_8.toString()
+
+        kotlinOptions {
+            jvmTarget = JavaVersion.VERSION_1_8.toString()
+            freeCompilerArgs = listOf("-Xopt-in=kotlin.RequiresOptIn")
+            languageVersion = "1.3"
+        }
+    }
```

Comment:
I'm unsure about this to be honest.
Specifically the `languageVersion` and the opt-in flag should stay inside the included build (and not in the outer build). I'm afraid they won't be passed over.

As a side note, I would refrain from adding those customization inside the `subprojects{}` block.

Reply:
Yeah I agree about the flag and the languageVersion - will move it back, but why would you refrain from adding it in the subprojects block at all?

Reply:
>  will move it back, but why would you refrain from adding it in the subprojects block at all?

Mostly because:
- This project is setup as a composite build, and `subprojects{}` is not running on included builds. That could be confusing for who's reading the build file.
- I would try to keep the compiler customization as closer as possible to the source code where they're needed. From this diff I can't say anymore why `-Xopt-in=kotlin.RequiresOptIn` is needed. While if we keep it closer to the module where it's effectively needed, it will make it easier to understand/cleanup afterwords.
- Generally `subprojects{}` creates a tight couplign between the projects. I think it's ok to use as long as you setup linter/formatter/test-runners or other tools that you effectively want to apply for every module. But as soon as you need to specify some build specific settings, you're losing on isolation as soon as you add configuration there

Reply:
As a side note: I believe we can leave the `sourceCompatibility` and `targetCompatibility` in the `subprojects{}` block as they should be applied to all the project uncoditionally.

Reply:
> This project is setup as a composite build, and subprojects{} is not running on included builds

because of that, I agree with @cortinico

Reply:
Yeah, it was clear for the compiler options where is their better place, but I didn't know that subprojects wasn't applied for composite builds. Thanks! Moved it back